src/Microsoft.DotNet.ProjectModel/ProjectReader.cs

```diff
@@ -246,6 +243,38 @@ public Project ReadProject(Stream stream, string projectName, string projectPath
             return project;
         }
 
+        internal static string NormalizeProjectDirectoryPath(string path)
+        {
+            if (File.Exists(path) &&
```

Comment:
It feels to me that the readability of this method could be improved a lot by simply extracting contents of the if conditions into nicely named methods. Likewise for the contents of the body of the ifs.